src/libMessage/Messenger.cpp

```diff
@@ -4349,14 +4349,39 @@ bool Messenger::SetNodeForwardTxnBlock(
   unsigned int txnsCurrentCount = 0;
   unsigned int txnsGeneratedCount = 0;
 
-  for (const auto& txn : txnsCurrent) {
-    TransactionToProtobuf(txn, *result.add_transactions());
+  unsigned int msg_size = 0;
+
+  for (auto iter = txnsCurrent.begin(); iter != txnsCurrent.end();) {
+    if (msg_size >= PACKET_BYTESIZE_LIMIT) {
+      break;
+    }
+    ProtoTransaction* protoTxn = new ProtoTransaction();
+    TransactionToProtobuf(*iter, *protoTxn);
+    unsigned txn_size = protoTxn->ByteSize();
+    if ((msg_size + txn_size) > PACKET_BYTESIZE_LIMIT &&
+        txn_size >= SMALL_TXN_SIZE) {
+      continue;
+    }
+    *result.add_transactions() = *protoTxn;
     txnsCurrentCount++;
+    msg_size += protoTxn->ByteSize();
+    iter = txnsCurrent.erase(iter);
   }
 
   for (const auto& txn : txnsGenerated) {
-    TransactionToProtobuf(txn, *result.add_transactions());
+    if (msg_size >= PACKET_BYTESIZE_LIMIT) {
+      break;
+    }
+    ProtoTransaction* protoTxn = new ProtoTransaction();
+    TransactionToProtobuf(txn, *protoTxn);
+    unsigned txn_size = protoTxn->ByteSize();
+    if ((msg_size + txn_size) > PACKET_BYTESIZE_LIMIT &&
```

Comment:
What is the logic here?

Reply:
first check is msg_size exceeded or not.
then we take a txn, to see if msg_size + this txn exceeded the limit,
if exceeded, if it's a very huge txn, we ignore it.
if it's a small txn, we accept it.
This can be used to prevent somebody intently put a very huge size txn to block the packet formation